File: Funciones.py
a=(1,2,3,4,5,)
b=(6,7,8,9,10)
c=[]

for contador in range(len(a)):
    c.append(a[contador]*b[contador])
print(c)    

#Funciones

#Definicion y llamada

def mostrar_texto():
    print('hola')

mostrar_texto()

def multiplicar():
    a = 5
    b = 8
    print(a*b)
multiplicar()

#Ambito de las variables, las variables declaradas dentro de la funcion no son accesibles desde afuera, solo desde la propia funcion
#Las de afuera sin embargo, si son accesibles desde dentro de la funcion.

def multiplicar():
    print(a*b)
a = 5
b = 8
multiplicar()

#Las funciones pueden devolver los valores con la instruccion return
#En el siguiente caso devolveremos la multiplicacion, y el resultado lo guardamos en una variable, que luego imprimimos sumandole un valor

def multiplicar():
    a = 5
    b = 8
    return a * b
Resultado=multiplicar()
print(Resultado+5)

#Podemos devolver cualquier tipo de dato, y tratarlo como tal fuera de la funcion

def validar_dato():
    if a==5:
        return True
    else:
        return False

a = 5
dato=validar_dato()
print(dato)

File: funciones.js
// Definir los arreglos
let a = [1, 2, 3, 4, 5];
let b = [6, 7, 8, 9, 10];
let c = [];

// Multiplicar elementos de los arreglos
for (let i = 0; i < a.length; i++) {
    c.push(a[i] * b[i]);
}
console.log(c);

// Funciones

// Definición y llamada
function mostrarTexto() {
    console.log("hola");
}
mostrarTexto();

function multiplicar() {
    let a = 5;
    let b = 8;
    console.log(a * b);
}
multiplicar();

// Ámbito de variables
function multiplicarConVariablesGlobales() {
    console.log(a * b);
}
let aGlobal = 5;
let bGlobal = 8;
multiplicarConVariablesGlobales();

// Función con retorno de valor
function multiplicarConRetorno() {
    let a = 5;
    let b = 8;
    return a * b;
}
let resultado = multiplicarConRetorno();
console.log(resultado + 5);

// Función que devuelve un booleano
function validarDato() {
    return a === 5;
}

let aBoolean = 5;
let dato = validarDato();
console.log(dato);
